platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java

```diff
@@ -199,8 +200,43 @@ private Browser(final Set<Integer> localNodesToStart) throws IOException {
                 .withConfigDataType(MetricsConfig.class)
                 .withConfigDataType(PrometheusConfig.class)
                 .withConfigDataType(OSHealthCheckConfig.class)
-                .withConfigDataType(WiringConfig.class)
-                .build();
+                .withConfigDataType(WiringConfig.class);
+
+        // The following variables are not final while being initialized.
+        ApplicationDefinition appDefinitionInit = null;
+        Map<Long, SwirldMain> appMainsInit = null;
+        Configuration config = null;
+        Exception exception = null;
+        try {
+            // Load config.txt file, parse application jar file name, main class name, address book, and parameters
+            appDefinitionInit = ApplicationDefinitionLoader.load(configurationProperties, localNodesToStart);
+
+            // Load all SwirldMain instances for locally run nodes.
+            appMainsInit = loadSwirldMains(appDefinitionInit, localNodesToStart);
+
+            // We assume all locally run instances would provide the same config to the configuration builder.
+            if (appMainsInit.size() > 0) {
+                appMainsInit.get(0L).updateConfigurationBuilder(configurationBuilder);
+            }
+
+            config = configurationBuilder.build();
+        } catch (final Exception e) {
+            logger.error(EXCEPTION.getMarker(), "Failed to load application due to exception.", e);
+            exception = e;
+        } finally {
+            // this.configuration needs to be initialized before the constructor returns or throws exception.
+            this.configuration = config;
+            if (exception != null) {
+                return;
```

Comment:
Why not throwing the exception?

Reply:
runtime exception is now being thrown in next commit.